Content.Server/Fax/FaxSystem.cs

```diff
@@ -0,0 +1,390 @@
+                if (isAnimationEnd)
+                {
+                    _itemSlotsSystem.SetLock(comp.Owner, comp.PaperSlot, false);
+                    UpdateUserInterface(comp.Owner, comp);
+                }
+            }
+
+            // Sending timeout
+            if (comp.SendTimeoutRemaining > 0)
+            {
+                comp.SendTimeoutRemaining -= frameTime;
+                
+                if (comp.SendTimeoutRemaining <= 0)
+                    UpdateUserInterface(comp.Owner, comp);
+            }
+        }
+    }
+
+    private void OnComponentInit(EntityUid uid, FaxMachineComponent component, ComponentInit args)
+    {
+        _itemSlotsSystem.AddItemSlot(uid, FaxSystem.PaperSlotId, component.PaperSlot);
+        UpdateAppearance(uid, component);
+    }
+
+    private void OnComponentRemove(EntityUid uid, FaxMachineComponent component, ComponentRemove args)
+    {
+        _itemSlotsSystem.RemoveItemSlot(uid, component.PaperSlot);
+    }
+
+    private void OnMapInit(EntityUid uid, FaxMachineComponent component, MapInitEvent args)
+    {
+        // Load all faxes on map in cache each other to prevent taking same name by user created fax
+        Refresh(uid, component);
+    }
+
+    private void OnItemSlotChanged(EntityUid uid, FaxMachineComponent component, ContainerModifiedMessage args)
+    {
+        if (!component.Initialized)
+            return;
+
+        if (args.Container.ID != component.PaperSlot.ID)
+            return;
+
+        var isPaperInserted = component.PaperSlot.Item.HasValue;
+        if (isPaperInserted)
+        {
+            component.InsertingTimeRemaining = component.InsertionTime;
+            _itemSlotsSystem.SetLock(uid, component.PaperSlot, true);
+        }
+
+        UpdateUserInterface(uid, component);
+    }
+    
+    private void OnInteractUsing(EntityUid uid, FaxMachineComponent component, InteractUsingEvent args)
+    {
+        if (args.Handled ||
+            !TryComp<ActorComponent>(args.User, out var actor) ||
+            !_toolSystem.HasQuality(args.Used, "Screwing")) // Screwing because Pulsing already used by device linking
+            return;
+
+        _quickDialog.OpenDialog(actor.PlayerSession,
+            Loc.GetString("fax-machine-dialog-rename"),
+            Loc.GetString("fax-machine-dialog-field-name"),
+            (string newName) =>
+        {
+            if (component.FaxName == newName)
+                return;
+
+            if (newName.Length > 20)
+            {
+                _popupSystem.PopupEntity(Loc.GetString("fax-machine-popup-name-long"), uid, Filter.Pvs(uid));
+                return;
+            }
+
+            if (component.KnownFaxes.ContainsValue(newName) && !component.Emagged) // Allow exist names if emagged for fun
+            {
+                _popupSystem.PopupEntity(Loc.GetString("fax-machine-popup-name-exist"), uid, Filter.Pvs(uid));
+                return;
+            }
+
+            args.Handled = true;
```

Comment:
This should be outside of the dialog code, the interaction gets handled the moment it has an effect (which in this case is opening the dialog UI).